Assets/MixedRealityToolkit/_Core/Inspectors/Profiles/MixedRealityControllerVisualizationProfileInspector.cs

```diff
@@ -0,0 +1,271 @@
+                    globalLeftHandModel.objectReferenceValue = leftHandModelPrefab;
+                }
+
+                EditorGUI.BeginChangeCheck();
+                rightHandModelPrefab = EditorGUILayout.ObjectField(new GUIContent(globalRightHandModel.displayName, "Note: If the default model is not found, the fallback is the global right hand model."), rightHandModelPrefab, typeof(GameObject), false) as GameObject;
+
+                if (EditorGUI.EndChangeCheck() && CheckVisualizer(rightHandModelPrefab))
+                {
+                    globalRightHandModel.objectReferenceValue = rightHandModelPrefab;
+                }
+
+                EditorGUIUtility.labelWidth = defaultLabelWidth;
+
+                RenderControllerList(controllerVisualizationSettings);
+            }
+
+            serializedObject.ApplyModifiedProperties();
+        }
+
+        private void RenderControllerList(SerializedProperty controllerList)
+        {
+            if (thisProfile.ControllerVisualizationSettings.Length != controllerList.arraySize) { return; }
+
+            EditorGUILayout.Space();
+
+            if (GUILayout.Button(ControllerAddButtonContent, EditorStyles.miniButton))
+            {
+                controllerList.InsertArrayElementAtIndex(controllerList.arraySize);
+                var index = controllerList.arraySize - 1;
+                var controllerSetting = controllerList.GetArrayElementAtIndex(index);
+                var mixedRealityControllerMappingDescription = controllerSetting.FindPropertyRelative("description");
+                mixedRealityControllerMappingDescription.stringValue = typeof(GenericJoystickController).Name;
+                var mixedRealityControllerHandedness = controllerSetting.FindPropertyRelative("handedness");
+                mixedRealityControllerHandedness.intValue = 1;
+                serializedObject.ApplyModifiedProperties();
+                thisProfile.ControllerVisualizationSettings[index].ControllerType.Type = typeof(GenericJoystickController);
+                return;
+            }
+
+            for (int i = 0; i < controllerList.arraySize; i++)
+            {
+                EditorGUILayout.Space();
+                EditorGUILayout.BeginHorizontal();
+
+                EditorGUIUtility.labelWidth = 64f;
+                EditorGUIUtility.fieldWidth = 64f;
+                var controllerSetting = controllerList.GetArrayElementAtIndex(i);
+                var mixedRealityControllerMappingDescription = controllerSetting.FindPropertyRelative("description");
+                bool hasValidType = thisProfile.ControllerVisualizationSettings[i].ControllerType != null &&
+                                    thisProfile.ControllerVisualizationSettings[i].ControllerType.Type != null;
+                if (hasValidType)
+                {
+                    mixedRealityControllerMappingDescription.stringValue = thisProfile.ControllerVisualizationSettings[i].ControllerType.Type.Name.ToProperCase();
```

Comment:
Not blockers, but:

1. We should reset this to some default description if `hasValidType` is false
2. There appears to be strange spacing in the name when this is rendered (note the larger spacing between the words in the controller type vs "Right Hand"

![image](https://user-images.githubusercontent.com/3580640/46445375-6f528400-c72b-11e8-8757-b9612b12518f.png)

Reply:
yeah I also noticed that the spacing was strange, but I'm not sure what's going on because I'm only adding a single space. Maybe Unity is increasing the word spacing in the label somehow?

Do you have any suggestions on a default description? I'm more than willing to accommodate that.  Maybe `Spatial Controller <handedness> Hand`?

Reply:
Yeah, that would work. Or, since we're notifying them that a controller type must be defined, even something like `Undefined Controller <handedness> Hand` seems fine. I'll leave it up to you, and it can always be iterated on in the future.

Reply:
>I'm not sure what's going on because I'm only adding a single space

Ah I figured it out! You're calling `ToProperCase()` both when you set the `stringValue` and when you turn the `stringValue` into the `LabelField`. Both calls are adding a space.

Reply:
Yup I just noticed that. I'll be sure to fix it as well.

Reply:
Fix incoming